### DISCUSSION TITLE
Can't Find  InterruptableJob

### DISCUSSION THREAD

**Discussion by SimpleJouer** [quartznet/quartznet#1435] [Q&A] [answered]
I need to  implement InterruptableJob  class to interrupt job.I can find this class  old version ,but no find int the latest version(3.3.3).What should I do

**Comment by lahma** [comment]
You need to observe a cancellation token, [it's documented in the migration guide](https://www.quartz-scheduler.net/documentation/quartz-3.x/migration-guide.html#packaging-changes).

**Reply by SimpleJouer** [reply]
I don't understand

**Reply by drieseng** [reply]
@lahma Can you explain why **IInterruptableJob** was removed? I would've liked to have the possibility of opting out of the extra cost that is involved (the instantiation of a **CancellationTokenSource** and a **CancellationToken**) if my job(s) cannot be interrupted.

Is this something we can discuss?

**Reply by lahma** [reply]
Because we are async it's the standard way of doing things. I don't see that much extra cost in one cancellation source?

**Reply by drieseng** [reply]
@lahma No, it's surely not much and as part of #1430 I'll only initialize it on demand anyway (if approved of course). Never mind.

**Comment by SimpleJouer** [comment, marked as answer]
how do I use  CancellationTokenSource???

**Reply by lahma** [reply]
https://github.com/quartznet/quartznet/tree/main/src/Quartz.Examples/07_InterrupInProgressJobs
